tests/e2e/server/common.hpp

```diff
@@ -0,0 +1,62 @@
+// Copyright 2022 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include <chrono>
+#include <functional>
+#include <thread>
+
+#include <spdlog/spdlog.h>
+#include <boost/asio/io_context.hpp>
+#include <boost/asio/steady_timer.hpp>
+#include <boost/system/detail/error_code.hpp>
+#include <mgclient.hpp>
+
+#include "utils/logging.hpp"
+
+inline void OnTimeoutExpiration(const boost::system::error_code &ec) {
+  if (!ec) {
```

Comment:
And you can remove the if :)

Reply:
Hmm this won't work since this callback is called on both expiration and cancellation and the only way to differentiate is to check if there is no error present, then we know we cancelled for sure.

Reply:
Yes, you are right. This will do the trick `MG_ASSERT(!!ec, "Connection timeout");`